release/kdm/rke2.go

```diff
@@ -0,0 +1,467 @@
+package kdm
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type (
```

Comment:
Can you break this into individual structs?

Reply:
done